ftd-host/auth.ftd

```diff
@@ -0,0 +1,121 @@
+-- import: fastn.com/ftd/built-in-variables as v
+-- import: fastn/processors as pr
+-- import: fastn.com/ftd-host/processor
+-- import: fastn.com/backend/env-vars
+-- import: fifthtry.github.io/admonitions as cbox
+
+-- ds.page: Github User Details using the `user-details` processor
+
+Let's look at how we can access basic user details of a user authenticated
+using GitHub.
+
+-- experimental-warning:
+-- processor.static-vs-dynamic:
+
+-- ds.markdown:
+
+We can use `user-details` processor to get information about the
+authenticated user. First, we need some data structures:
+
+-- ds.code: Types required by `user-details`
+lang: ftd
+
+\-- record status:
+boolean is-login:
+optional user-details user:
+
+\-- record user-details:
+string name:
+string login:
```

Comment:
Is this username? Let's call it so.

Reply:
yes, `login` is username. Github api [calls it `login`](https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-the-authenticated-user) which is why I decided to not name it `username`. Do you still want me to rename this to `username`? This will require changing the rust code for this.